### DISCUSSION TITLE
Example model config yaml?

### DISCUSSION THREAD

**Discussion by mattcaron** [LykosAI/StabilityMatrix#1407] [Q&A]
Hello folks,

I've found the help which says that "You can use a config (.yaml) file to load a model with specific settings" and it tells you where it goes.

It does not, however, tell you the format of the yaml file (besides that it's a yaml file, of course).

Does anyone know where I might find an example or skeleton model config `.yaml` file?

Thanks in advance.

**Comment by xen0form** [comment]
Same question.
Tried following:

Sampler: DPM++ SDE
Scheduler: Karras
Steps: 25

But no luck, Inference doesn't recall it.

**Comment by xWris3** [comment]
Hi all,

Same "problem" on my side. I wonder if this is an incomplete feature or something.
It is completely undocumented, so I don't know what the correct keys are for the yaml (although I guess I will end-up looking at the code).
Anyways, once I create a yaml file next to the checkpoint, I see a little lab bottle (apologies for my poor chemistry terminology) in the GUI, but it doesn't really change anything. It recognizes there is a file, but that's about it.

<img width="102" height="33" alt="image" src="https://github.com/user-attachments/assets/ec47d590-ef6b-481f-a868-598e4f31d007" />

I thought maybe changing the setting there in the "Checkpoints manager":
<img width="1037" height="400" alt="image" src="https://github.com/user-attachments/assets/89932f7e-6ee1-4837-a292-73a67b38a9d5" />
would perhaps write the correct keys to the file but that's not the case.

In the YAML I tried this:
```
sampler: "Euler Ancestral"
scheduler: "SGM Uniform"
steps: 30
cfg_scale: 5.0
width: 1024
height: 1156
```
But no luck.

**Comment by xWris3** [comment]
I did some quick digging.

looking at the code in this repo, it seems it is handled here (StabilityMatrix.Avalonia\ViewModels\Inference\ModelCardViewModel.cs):
```
    private static ComfyTypedNodeBase<
        ModelNodeConnection,
        ClipNodeConnection,
        VAENodeConnection
    > GetDefaultModelLoader(ModuleApplyStepEventArgs e, string nodeName, HybridModelFile model)
    {
        // Check if config
        if (model.Local?.ConfigFullPath is { } configPath)
        {
            // We'll need to upload the config file to `models/configs` later
            var uploadConfigPath = e.AddFileTransferToConfigs(configPath);

            return new ComfyNodeBuilder.CheckpointLoader
            {
                Name = nodeName,
                // Only the file name is needed
                ConfigName = Path.GetFileName(uploadConfigPath),
                CkptName = model.RelativePath,
            };
        }

        // Simple loader if no config
        return new ComfyNodeBuilder.CheckpointLoaderSimple { Name = nodeName, CkptName = model.RelativePath };
    }
```
and also (StabilityMatrix.Core\Models\Api\Comfy\Nodes\ComfyNodeBuilder.cs):
```
    public record CheckpointLoader
        : ComfyTypedNodeBase<ModelNodeConnection, ClipNodeConnection, VAENodeConnection>
    {
        public required string ConfigName { get; init; }
        public required string CkptName { get; init; }
    }

    public record CheckpointLoaderSimple
        : ComfyTypedNodeBase<ModelNodeConnection, ClipNodeConnection, VAENodeConnection>
    {
        public required string CkptName { get; init; }
    }
```

So this is to use/call comfyui nodes, as a client (StabilityMatrix is a client of the running Comfyui).
Looking in Comfyui source, I find it seems the "CheckpointLoader" with a config file is now deprecated (see: https://github.com/search?q=repo%3AComfy-Org%2FComfyUI+nodes.py+CheckpointLoader&type=code).

I have no clue if this means calling this node falls back to the "Simple" variant (which doesn't support config file). If it does, it would explain the behavior.
If not, we just don't know the syntax expected.

**Reply by ionite34** [reply]
Hi, the model yaml config file can look something like this (vpred example).

It's usually given by the model authors if there is some non-standard inference settings that can't otherwise be adjusted in the UI.

```yaml
model:
  base_learning_rate: 1.0e-04
  target: ldm.models.diffusion.ddpm.LatentDiffusion
  params:
    parameterization: "v"
    linear_start: 0.00085
    linear_end: 0.0120
    num_timesteps_cond: 1
    log_every_t: 200
    timesteps: 1000
    first_stage_key: "jpg"
    cond_stage_key: "txt"
    image_size: 64
    channels: 4
    cond_stage_trainable: false
    conditioning_key: crossattn
    monitor: val/loss_simple_ema
    scale_factor: 0.18215
    use_ema: False

    scheduler_config: # 10000 warmup steps
      target: ldm.lr_scheduler.LambdaLinearScheduler
      params:
        warm_up_steps: [ 10000 ]
        cycle_lengths: [ 10000000000000 ]
        f_start: [ 1.e-6 ]
        f_max: [ 1. ]
        f_min: [ 1. ]

    unet_config:
      target: ldm.modules.diffusionmodules.openaimodel.UNetModel
      params:
        image_size: 32 # unused
        in_channels: 4
        out_channels: 4
        model_channels: 320
        attention_resolutions: [ 4, 2, 1 ]
        num_res_blocks: 2
        channel_mult: [ 1, 2, 4, 4 ]
        num_heads: 8
        use_spatial_transformer: True
        transformer_depth: 1
        context_dim: 768
        use_checkpoint: True
        legacy: False

    first_stage_config:
      target: ldm.models.autoencoder.AutoencoderKL
      params:
        embed_dim: 4
        monitor: val/rec_loss
        ddconfig:
          double_z: true
          z_channels: 4
          resolution: 256
          in_channels: 3
          out_ch: 3
          ch: 128
          ch_mult:
          - 1
          - 2
          - 4
          - 4
          num_res_blocks: 2
          attn_resolutions: []
          dropout: 0.0
        lossconfig:
          target: torch.nn.Identity

    cond_stage_config:
      target: ldm.modules.encoders.modules.FrozenCLIPEmbedder
```

**Reply by xWris3** [reply]
thank you for your anwser !

I realized after having that "lab bottle" I talked about in the GUI, my model wouldn't start generating image (wouldn't load anymore), because of the malformed yaml. So thanks a lot for this.
But your answer makes me realize this config file is not for what I think me and the other people here thought it was used for.
I thought it was a way to define the "Inference Defaults":
<img width="1044" height="424" alt="image" src="https://github.com/user-attachments/assets/b480d098-9352-4fdd-a58e-060123a3285c" />
but that's not what this is it seems. Because this GUI option is not saving anything in my experience. Despite configuring something and enabling the switch, creating a new project then selecting the checkpoint on which I set the "Inference Defaults", nothing is saved.
